cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -449,6 +449,30 @@ TYPED_TEST(TestStringKernels, Utf8Reverse) {
   ASSERT_TRUE(res->array()->buffers[1]->Equals(*malformed_input->data()->buffers[1]));
 }
 
+TYPED_TEST(TestStringKernels, Utf8Nfd) {
+  this->CheckUnary("utf8_nfd", "[]", this->type(), "[]");
+  this->CheckUnary("utf8_nfd", R"(["①②3", null, "", "áéíﬁ 0😀"])", this->type(),
+                   R"(["①②3", null, "", "áéíﬁ 0😀"])");
+}
+
+TYPED_TEST(TestStringKernels, Utf8Nfkd) {
+  this->CheckUnary("utf8_nfkd", "[]", this->type(), "[]");
+  this->CheckUnary("utf8_nfkd", R"(["①②3", null, "", "áéíﬁ 0😀"])", this->type(),
+                   R"(["123", null, "", "áéífi 0😀"])");
+}
+
+TYPED_TEST(TestStringKernels, Utf8Nfc) {
+  this->CheckUnary("utf8_nfc", "[]", this->type(), "[]");
+  this->CheckUnary("utf8_nfc", R"(["①②3", null, "", "áéíﬁ 0😀"])", this->type(),
+                   R"(["①②3", null, "", "áéíﬁ 0😀"])");
+}
```

Comment:
It would be good to have test cases that do change input string when normalized. Similarly, for `utf8_nfd`.